web-app/src/app/app/jobs/[agentId]/@right/[jobId]/page.tsx

```diff
@@ -0,0 +1,39 @@
+import { notFound } from "next/navigation";
+
+import { JobDetails } from "@/components/job/job-details";
+import { requireAuthentication } from "@/lib/auth/utils";
+import { getAgentById } from "@/lib/db/services/agent.service";
+import { getJobById } from "@/lib/db/services/job.service";
+
+interface JobPageParams {
+  agentId: string;
+  jobId: string;
+}
+
+export default async function JobDetailPage({
+  params,
+}: {
+  params: Promise<JobPageParams>;
+}) {
+  const { agentId, jobId } = await params;
+
+  const agent = await getAgentById(agentId);
+  if (!agent) {
+    console.log("agent not found in job detail page");
+    notFound();
+  }
+
+  const job = await getJobById(jobId);
+  if (!job) {
+    notFound();
+  }
+  if (job.agent.id !== agentId) {
+    notFound();
+  }
+  const { session } = await requireAuthentication();
+  if (job.userId !== session.user.id) {
+    notFound();
+  }
```

Comment:
We take care of the error, but it would also be nice to take care of a loading state. As we have usually done it. A `loading.tsx` would be the easiest solution.

Reply:
Resolved in https://github.com/masumi-network/sokosumi/pull/248/commits/b8729c190c2f55cc1cd21eaadb87a1c77bec8b14